pennylane/operation.py

```diff
@@ -480,6 +480,67 @@ def matrix(self, wire_order=None):
 
         return expand_matrix(canonical_matrix, wires=self.wires, wire_order=wire_order)
 
+    @staticmethod
+    def compute_sparse_matrix(*params, **hyperparams):  # pylint:disable=unused-argument
+        """Canonical matrix of this operator in the computational basis, using a sparse
+        matrix type.
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. note::
+            This method gets overwritten by subclasses to define the sparse matrix representation
+            of a particular operator.
+
+        Args:
+            params (list): trainable parameters of this operator, as stored in ``op.parameters``
+            hyperparams (dict): non-trainable hyperparameters of this operator, as stored in ``op.hyperparameters``
+
+        Returns:
+            scipy.sparse.coo.coo_matrix: matrix representation
+
+        **Example**
+
+
+        """
+        raise NotImplementedError
+
+    def sparse_matrix(self, wire_order=None):
+        r"""Matrix representation of this operator in the computational basis, using
+         a sparse matrix type.
+
+         If ``wire_order`` is provided, the
+         numerical representation considers the position of the
+         operator's wires in the global wire order.
+         Otherwise, the wire order defaults to ``self.wires``.
+
+         .. note::
+             By default, this method calls the static method ``compute_sparse_matrix``,
+             which is used by subclasses to define the actual numerical representation.
+
+         A ``NotImplementedError`` is raised if the matrix representation has not been defined.
+
+         .. note::
+             The wire_order argument is not yet implemented.
+
+         Args:
+             wire_order (Iterable): global wire order, must contain all wire labels from this operator's wires
+
+         Returns:
+             scipy.sparse.coo.coo_matrix: matrix representation
+
+         **Example**
+
+        XXX
+
+        """
+        if wire_order is not None:
+            raise NotImplementedError("The wire_order argument is not yet implemented")
```

Comment:
how come?

Reply:
I have no clue how to rewrite the `expand` method for sparse matrices. If anyone finds this error I'll do it, promise!